force-app/main/default/classes/ConnectApiHelper.cls

```diff
@@ -197,12 +200,24 @@ global class ConnectApiHelper {
                         // Add a segment for any accumulated text (which includes unsupported HTML tags).
                         addTextSegment(messageSegmentInputs, textSegment);
 
-                        ConnectApi.MarkupBeginSegmentInput markupBeginSegmentInput = makeMarkupBeginSegmentInput(tag);
+                        ConnectApi.MarkupBeginSegmentInput markupBeginSegmentInput = makeMarkupBeginSegmentInput(tag, null, null);
                         messageSegmentInputs.add(markupBeginSegmentInput);
                         strPos = globalMatcher.end();
+                    } else if (tag.toLowerCase().contains('href=')){
+                        // Add a segment for any accumulated text (which includes unsupported HTML tags).
+                        addTextSegment(messageSegmentInputs, textSegment);
+                        //hyperlink tag - get the URL. Assumes format of <a href="url">altText</a>
+                        String url = matchingText.substring(matchingText.indexOf('"') + 1, matchingText.lastIndexOf('>') - 1);
```

Comment:
```suggestion
                        String url = matchingText.substring(matchingText.indexOf('"') + 1, matchingText.indexOf('>') - 1);
```
If we use lastIndexOf it will return the URL plus everything after it until `</`